### PR TITLE
Use lvm/btrfs for images imported via launch remote

### LINKED ISSUE #805: 'lxc launch lxc:ubuntu/trusty/amd64 t1' with LVM storage does not create an LV
there is a separate code path for downloading and importing the image specified in the launch command that skips the checks for LVM and btrfs.

### PR DESCRIPTION
**Pull request by mikemccracken**
Fixes #805

Ensure that images imported as a result of 'lxc launch remote:imagename'
are using the correct storage backend by sharing a new function
'buildImageFromInfo' with imagesPost.

Adds a test for this to test_lvm

Signed-off-by: Michael McCracken <mike.mccracken@canonical.com>

### CODE REVIEW

**Comment by mikemccracken** [comment]
Note for reviewers - this changes the SQL for inserting the image row in the default import path: https://github.com/lxc/lxd/pull/808/files#diff-a920a20c1236ff4ae4d8cf57be5f1150L461

Previously, creation_date and expiry_date were not being set on images imported via 'lxc image import ...'

File: test/lvm.sh [hunk 1]
```diff
@@ -39,7 +39,10 @@ die() {
 }
 
 test_lvm() {
-    which vgcreate || echo "===> SKIPPING lvm backing: vgcreate not found" && return
+    if [ ! `which vgcreate` ]; then
```
Comment (stgraber) [inline test/lvm.sh]: No need to run through test here.

    if ! which vgcreate >/dev/null; then

Will save you a fork()

**Comment by stgraber** [comment]
Can there be a case where vgcreate is available but lvs isnt?

**Comment by mikemccracken** [comment]
@stgraber no, vg* , lv* , pv* are all in pkg lvm2.

**Comment by stgraber** [comment]
@mikemccracken even on the redhat based distros?

**Comment by mikemccracken** [comment]
@stgraber good question, I had to look - looks like yes: https://apps.fedoraproject.org/packages/lvm2/contents/

**Comment by stgraber** [comment]
Perfect then

**Pull request merged**